### PR TITLE
[F] Gallery Tiles can use images instead of backgrounds

### PR DESCRIPTION
**Pull request by drapisarda**
With a parameter, we can have images elements instead of backgrounds for HdGalleryTiles

### CODE REVIEW

**Comment by SinisaG** [comment]
Agree, please add tests for `srcSet`. It will be very helpful, as we use this a lot.

**Comment by drapisarda** [comment]
In HDLazyImage, I'm trying to test the Image load. My issue is the t the loading is detected using an Image object (:62) that I can not access via jest. Any idea on how to do this? 

Probably the Image object should accessible from the outer scope.

**Comment by SinisaG** [comment]
@drapisarda : One way that comes to mind is to expose the reference to the image. And then call `onload` manually. 

e.g.: 

```javascript
  ....
  data() {
    return {
      imageLoaded: false,
      img: null,
    };
  },
 ....
    maybeLazyLoad() {
      this.setImageUrl('');

      if (!this.srcSmall) {
        this.setImageUrl(this.src);
        this.imageLoaded = true;
        return;
      }

      this.imageLoaded = false;
      this.setImageUrl(this.srcSmall);

      this.img = new Image();

      this.img.onload = () => {
        this.imageLoaded = true;
        this.setImageUrl(this.src, this.srcset);
      };

      this.img.src = this.src;
    },
```

And your test could look something like that: 
```javascript
  it('Test onload callback', async () => {
    wrapper.vm.img.onload()
    console.log(wrapper.html());
  });
```

I'll think about possible alternatives way. But this is pretty simple and not overwriting anything. I guess, as you'd like to tests the outcome of `onload` that could be fine.

**Comment by drapisarda** [comment]
I removed all the srcset part to keep this PR scoped.

File: tests/unit/components/HdLazyImage.spec.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    });
+  });
+
+  it('renders component', () => {
+    expect(wrapper.html()).toMatchSnapshot();
+  });
+
+  it('It fist display the thumbnail/small image', () => {
+    expect(wrapper.attributes().src).toBe(sampleImage.thumbnail);
+  });
```
Comment (viniciuskneves) [inline tests/unit/components/HdLazyImage.spec.js]: Aren't those tests similar? I think you can condense both in just one test: snapshot + src.

Comment (drapisarda) [reply tests/unit/components/HdLazyImage.spec.js]: Yes. I could actually just keep the toMatchSnapshot one.

**Pull request merged**